code/game/machinery/vitals_monitor.dm

```diff
@@ -13,12 +16,31 @@
 	construct_state = /decl/machine_construction/default/panel_closed
 
 	var/mob/living/carbon/human/victim
+	var/obj/machinery/optable/connected_optable = null
 	var/beep = TRUE
 
+	//alert stuff
+	var/read_alerts = TRUE
+	var/alert_cooldown = 50 // buffer to prevent rapid changes in state
+	var/last_alert_time = 0
+	var/list/alerts = null
+	var/list/last_alert = null
+
+/obj/machinery/vitals_monitor/Initialize()
+	. = ..()
+	alerts = new /list(3)
+	last_alert = new /list(3)
+	for(dir in list(NORTH,EAST,SOUTH,WEST))
+		connected_optable = locate(/obj/machinery/optable, get_step(src, dir))
+		if (connected_optable)
```

Comment:
Should check if it already has a monitor for mapping sanity purposes.

Reply:
👍